soroban-sdk/src/events.rs

```diff
@@ -112,3 +115,43 @@ impl Events {
         internal::Env::contract_event(env, topics.into_val(env).to_object(), data.into_val(env));
     }
 }
+
+#[cfg(feature = "testutils")]
+use crate::{xdr, TryIntoVal};
+
+#[cfg(feature = "testutils")]
+#[cfg_attr(feature = "docs", doc(cfg(feature = "testutils")))]
+impl Events {
+    /// Returns all events that have been published by contracts.
+    ///
+    /// Returns a [`Vec`] of three element tuples containing:
+    /// - Contract ID
+    /// - Event Topics as a [`Vec<RawVal>`]
+    /// - Event Data as a [`RawVal`]
+    pub fn all(&self) -> Vec<(crate::BytesN<32>, Vec<RawVal>, RawVal)> {
+        let env = self.env();
+        let mut vec = Vec::new(env);
+        self.env()
+            .host()
+            .get_events()
+            .unwrap()
+            .0
+            .into_iter()
+            .for_each(|e| {
+                if let internal::events::HostEvent::Contract(xdr::ContractEvent {
+                    type_: xdr::ContractEventType::Contract,
+                    contract_id: Some(contract_id),
+                    body: xdr::ContractEventBody::V0(xdr::ContractEventV0 { topics, data }),
+                    ..
+                }) = e
+                {
+                    vec.push_back((
+                        contract_id.0.into_val(env),
+                        topics.try_into_val(env).unwrap(),
+                        data.try_into_val(env).unwrap(),
+                    ))
```

Comment:
@jonjove I know we had some doubts on whether the ScVal <=> RawVal conversions were worth it, but they do allow for some nice conveniences in the SDK. Check this out ☝🏻 where we can do relatively seamless conversions from ScVal types to SDK types.